src/main/kotlin/com/terraformation/backend/tracking/db/T0PlotStore.kt

```diff
@@ -251,6 +252,65 @@ class T0PlotStore(
     )
   }
 
+  fun assignT0TempZoneSpeciesDensities(
+      plantingZoneId: PlantingZoneId,
+      densities: List<SpeciesDensityModel>,
+  ) {
+    requirePermissions { updateT0(plantingZoneId) }
+
+    val now = clock.instant()
+    val currentUserId = currentUser().userId
+
+    dslContext.transaction { _ ->
+      with(PLANTING_ZONE_T0_TEMP_DENSITIES) {
+        dslContext
+            .deleteFrom(this)
+            .where(PLANTING_ZONE_ID.eq(plantingZoneId))
+            .and(SPECIES_ID.notIn(densities.map { it.speciesId }))
+            .execute()
+
+        var insertQuery =
+            dslContext.insertInto(
+                this,
+                PLANTING_ZONE_ID,
+                SPECIES_ID,
+                ZONE_DENSITY,
+                CREATED_BY,
+                CREATED_TIME,
+                MODIFIED_BY,
+                MODIFIED_TIME,
+            )
+
+        densities.forEach {
+          if (it.density < BigDecimal.ZERO) {
+            throw IllegalArgumentException("Zone density must not be negative")
+          }
+          insertQuery =
+              insertQuery.values(
+                  plantingZoneId,
+                  it.speciesId,
```

Comment:
A named lambda parameter would be a little easier to follow than `it` since the `forEach` is several lines above this.